### ISSUE TITLE
Crescendo Hairpins - Import from MS4 failed

### ISSUE THREAD

**Issue by HeinzVo** [Jojo-Schmitz/MuseScore#175]
### Issue type

File corruption

### Bug description

Introduction:
-------------
Importing MS4.x scores to Mu3.7 is a very interesting feature, but at that early implementation state
a) a load error message pops up and must be clicked away
b) many elements and some styles go lost during import
c) the page layout gets corrupt due to erroneous rescaled element positions and spacer/gap-distances (should be 1=1)
   NOTE - page layout corruption problems are already reported here:
   https://github.com/Jojo-Schmitz/MuseScore/issues/165
   https://github.com/Jojo-Schmitz/MuseScore/issues/141
   https://github.com/musescore/MuseScore/issues/19014

Due to the corrupted page layout in MS3.7, it is difficult to reference to a page or a point in the attached score-examples, where a problem has detected.
First referencing the section/frame number in the score and then referencing a measure number in that section may identify the place (or just watch the screenshots).

New Reports, import from MS4 failes [->]:
-----------------------------------------
-  MuseScore 3.7.0_Staff Texts Style Expression - Import from MS4 failed  [Missa brevis in G-major, Section 6, Measure 159]
=> MuseScore 3.7.0_Crescendo Hairpins - Import from MS4 failed  [Missa brevis in G-major, Section 6, Measure 155]
-  MuseScore 3.7.0_Ornaments Short Trills - Import from MS4 failed  [Missa brevis in G-major, Section 6, Measures 134,142,148,151]
-  MuseScore 3.7.0_Ornaments Trills - Import from MS4 failed  [Gloria in D-major, Section 10, Measures 120,130]
-  and probably many other elements not yet tested ...

-  MuseScore 3.7.0_Crescendo Lines - Import from MS4 incomplete (Lines style wrong and not visible)  [Gloria in D-major, Section 12, Measures 76-77]
-  MuseScore 3.7.0_Text Lines - Import from MS4 incomplete (Lines style wrong and not visible)  [Gloria in D-major, Section 12, Measures 76-77]
-  MuseScore 3.7.0_Elements Style Dynamic - Import from MS4 incomplete (text style and size wrong)  [Missa brevis in G-major, Section 6, Measures 131,139,143,153,157]
-  and probably some other styles not yet tested ...

Other Reports, corrupting the page layout [https://github.com/Jojo-Schmitz/MuseScore/issues/165],[https://github.com/Jojo-Schmitz/MuseScore/issues/141]:
-----------------------------------------
-  MuseScore 3.7.0_Elements Offset - Import from MS4 corrupted (numbers changed)  [Missa brevis in G-major, Section 6, Measure 131]
-  MuseScore 3.7.0_Spacers and frame top¦bottom gaps - Import from MS4 corrupted (numbers changed)  [Gloria in D-major, Section 12, Measures 68/71,71/75]
-  MuseScore 3.7.0_Not hiding staff with subsequent empty measures after crescendo line in measure before  [Gloria in D-major, Section 10, Measures 1-8]

### Steps to reproduce

Steps to reproduce:
-------------------
1.) take a MS3.6.2-score and import it to MS4.x; save it there
2.) take that MS4.x-score and import it back to MS3.7

NOTE: check attached testscores and screenshots

### Screenshots/Screen recordings

![MuseScore 3 6 2_Crescendo Hairpins](https://github.com/Jojo-Schmitz/MuseScore/assets/140117374/8b5c784f-ff79-46de-8670-41f046adc5e2)
![MuseScore 3 7 0_Crescendo Hairpins - Import from MS4 failed](https://github.com/Jojo-Schmitz/MuseScore/assets/140117374/a9afee15-acbd-40b3-a75a-3a74969d5a29)

### MuseScore Version

MuseScore-3.7.0.6145742308-x86_64

### Regression

No.

### Operating system

Windows10 and 11, 64-bit

### Additional context

[Mu3.7 import from Mu4.x failed, incomplete or corrupt.zip](https://github.com/Jojo-Schmitz/MuseScore/files/12683820/Mu3.7.import.from.Mu4.x.failed.incomplete.or.corrupt.zip)

**Comment by Jojo-Schmitz** [comment]
Hairpins indeed get lost entirely on the round trip :-(

They are getting encoded exactly the same way in either version:
```xml
          <Spanner type="TextLine">
            <prev>
              <location>
                <measures>-1</measures>
                </location>
              </prev>
            </Spanner>
          <Spanner type="HairPin">
            <HairPin>
              <subtype>0</subtype>
              </HairPin>
            <next>
              <location>
                <measures>1</measures>
                </location>
              </next>
            </Spanner>
...

 ```
I'm currently at a loss to explain whey they get lost

**Comment by Jojo-Schmitz** [comment]
Thewre are a few new style settings in Mu4 with no counterparts in M3:
```
Debug: line 200 col 24: hairpinDashLineLen (...\libmscore\xmlreader.cpp:211, void Ms::XmlReader::unknown())
Debug: line 201 col 23: hairpinDashGapLen (...\libmscore\xmlreader.cpp:211, void Ms::XmlReader::unknown())
Debug: line 203 col 28: hairpinLineDashLineLen (...\libmscore\xmlreader.cpp:211, void Ms::XmlReader::unknown())
Debug: line 204 col 27: hairpinLineDashGapLen (...\libmscore\xmlreader.cpp:211, void Ms::XmlReader::unknown())
...
Debug: line 1393 col 26: hairpinShowTabSimple (...\libmscore\xmlreader.cpp:211, void Ms::XmlReader::unknown())
Debug: line 1394 col 26: hairpinShowTabCommon (...\libmscore\xmlreader.cpp:211, void Ms::XmlReader::unknown())
```

**Comment by Jojo-Schmitz** [comment]
#178 is another variant of the same issue apparently

**Comment by Jojo-Schmitz** [comment]
Indeed, the `lineStyle` changed fron numeric values to texts. So a simple remapping should do

Mu3 | Mu4
------|-----
2 | dashed
3 | dotted
5 | dotted (?)

TBC...

**Comment by Jojo-Schmitz** [comment]
Voltas, Let Ring lines, Ottavas and Palm Mute lines to be affected by this too
(also Tempo Change lines, but those don't exist in Mu3)
Pedal lines are not, but I'm not (yet) sure why...

**Comment by Jojo-Schmitz** [comment]
See #183

**Comment by Jojo-Schmitz** [comment]
Fixed via #183
